File: notes_and_examples/assert_vs_verify.py
# Assert example
def test_assert(expected, actual):
    assert expected == actual, f"Assertion failed: Expected '{expected}' but got '{actual}'"
    print("Assertion passed. Test continues...")

# Verify-like behavior using try-except
def test_verify(expected, actual):
    try:
        if expected != actual:
            raise AssertionError(f"Verification failed: Expected '{expected}' but got '{actual}'")
        print("Verification passed. Test continues...")
    except AssertionError as e:
        print(e)
        # Test can continue despite the failure

# Test cases
test_assert(5, 5)  # This assertion will pass
test_assert(5, 6)  # This assertion will fail

test_verify(10, 10)  # This verification will pass
test_verify(10, 11)  # This verification will fail, but the test will continue


"""

This code includes two functions: test_assert and test_verify.

test_assert uses the assert statement to perform an assertion. 

If the assertion fails, it raises an AssertionError and stops the execution immediately.

test_verify demonstrates a "verify"-like behavior using a try-except block. 

It checks a condition and, if the condition is not met, raises an AssertionError. 
However, it catches the error using the except block and allows the test to continue running.

"""

File: notes_and_examples/assert_vs_verify.java
public class AssertionExample {

    // Assert example
    public static void testAssert(int expected, int actual) {
        assert expected == actual : "Assertion failed: Expected " + expected + " but got " + actual;
        System.out.println("Assertion passed. Test continues...");
    }

    // Verify-like behavior using try-catch
    public static void testVerify(int expected, int actual) {
        try {
            if (expected != actual) {
                throw new AssertionError("Verification failed: Expected " + expected + " but got " + actual);
            }
            System.out.println("Verification passed. Test continues...");
        } catch (AssertionError e) {
            System.out.println(e.getMessage());
            // Test can continue despite the failure
        }
    }

    public static void main(String[] args) {
        testAssert(5, 5); // This assertion will pass
        testAssert(5, 6); // This assertion will fail

        testVerify(10, 10); // This verification will pass
        testVerify(10, 11); // This verification will fail, but the test will continue
    }
}

/*

 This Java code includes two methods: testAssert and testVerify.
testAssert uses the assert statement to perform an assertion. 
If the assertion fails, it throws an AssertionError and stops the execution immediately.

testVerify demonstrates a "verify"-like behavior using a try-catch block.
It checks a condition and, if the condition is not met, throws an AssertionError.
However, it catches the error and allows the test to continue running.

 */